source/kubernetes/minio-operator-reference.rst

```diff
@@ -104,94 +127,689 @@ The MinIO Operator adds a
+
+   - ``Never`` - Never pull the image.
+
+   - ``IfNotPresent`` - Pull the image if not already present.
+
+   Defaults to ``IfNotPresent`` if omitted.
+
+.. kubeconf:: spec.imagePullSecret
+
+   The secret to use for pulling images from private Docker repositories. 
+
+.. kubeconf:: spec.mountPath
+
+   *Optional*
+
+   The mount path for Persistent Volumes bound to :mc:`minio` pods in the
+   MinIO Tenant.
+
+   Defaults to ``/export``.
+
+.. kubeconf:: spec.securityContext
+
+   *Optional*
+
+   Root field for configuring the 
+   :kube-docs:`Security Context 
+   <tasks/configure-pod-container/security-context>` of pods created as part of
+   the MinIO Tenant.
+
+   The MinIO Operator supports the following 
+   :kube-api:`PodSecurityContext <#podsecuritycontext-v1-core>` fields:
+
+   - ``fsGroup``
+   - ``fsGroupChangePolicy``
+   - ``runAsGroup``
+   - ``runAsNonRoot``
+   - ``runAsUser``
+   - ``seLinuxOptions``
+
+   .. include:: /includes/common-minio-kubernetes.rst
+      :start-after: start-kubeapi-securitycontext
+      :end-before: end-kubeapi-securitycontext
+
+.. kubeconf:: spec.podManagementPolicy
+
+   *Optional*
+
+   The :kubedocs:`Pod Management Policy 
+   <concepts/workloads/controllers/statefulset/#pod-management-policies>` used
+   for pods created as part of the MinIO Tenant.
+
+   .. include:: /includes/common-minio-kubernetes.rst
+      :start-after: start-kubeapi-podmanagementpolicy
+      :end-before: end-kubeapi-podmanagementpolicy
+
+.. kubeconf:: spec.priorityClassName
+
+   *Optional*
+
+   The Pod :kube-docs:`Priority Class 
+   <concepts/configuration/pod-priority-preemption/#priorityclass>` to apply
+   to pods created as part of the MinIO Tenant.
+
+
+   .. include:: /includes/common-minio-kubernetes.rst
+      :start-after: start-kubeapi-priorityclassname
+      :end-before: end-kubeapi-priorityclassname
+
+.. kubeconf:: spec.s3
+
+   *Optional*
+
+   The S3-related features enabled on the MinIO Tenant.
+
+   Specify any of the following supported features as part of the 
+   :kubeconf:`~spec.s3` object:
+
+   - ``bucketDNS: <boolean>`` - specify ``true`` to enable DNS lookup of
+     buckets on the MinIO Tenant.
+
+.. kubeconf:: spec.serviceAccountName
+
+   *Optional*
+
+   The name of the 
+   :kube-docs:`Service Account 
+   <reference/access-authn-authz/service-accounts-admin/>` used to run all 
+   MinIO server :mc:`minio` pods created as part of the Tenant.
+
+.. kubeconf:: spec.subPath
+
+   *Optional*
+
+   The sub path appended to the :kubeconf:`spec.mountPath`. The resulting
+   full path is the directory in which MinIO stores data.
+
+   For example, given a :kubeconf:`~spec.mountPath` of ``export`` and
+   a :kubeconf:`~spec.subPath` of ``minio``, the full mount path is
+   ``export/minio``.
+
+   Defaults to empty (``""``).
+
+.. kubeconf:: spec.zones
+
+   *Required*
+
+   The configuration for each MinIO Zone deployed in the MinIO Tenant.
+
+
+
+
+     
+
+Transport Layer Encryption (TLS)
```

Comment:
@nitisht do any of the certificate related fields have any effect if `spec.autoCert` is 'disabled' or omitted?